web_widget_map/static/src/components/widget_map.esm.js

```diff
@@ -0,0 +1,137 @@
+/** @odoo-module */
+
+import {loadCSS, loadJS} from "@web/core/assets";
+import {registry} from "@web/core/registry";
+import {useService} from "@web/core/utils/hooks";
+import {standardFieldProps} from "@web/views/fields/standard_field_props";
+import {useInputField} from "@web/views/fields/input_field_hook";
+import {
+    Component,
+    onMounted,
+    onPatched,
+    onWillPatch,
+    onWillStart,
+    onWillUpdateProps,
+    useRef,
+    useState,
+} from "@odoo/owl";
+
+export class MapField extends Component {
+    static template = "map_field.MapField";
+    static props = {
+        ...standardFieldProps,
+    };
+
+    setup() {
+        this.rpc = useService("rpc");
+        this.orm = useRef("orm");
+        this.marker = null;
+        this.mapElementList = null;
+        this.state = useState({
+            location: {
+                lat: this.props.record.data[this.props.name]
+                    ? parseFloat(this.props.record.data[this.props.name].split(",")[0])
+                    : 0.0,
+                lng: this.props.record.data[this.props.name]
+                    ? parseFloat(this.props.record.data[this.props.name].split(",")[1])
+                    : 0.0,
+            },
+            fields: this.props.field_list
+                ? JSON.parse(this.props.field_list.split(","))
+                : [],
+        });
+        useInputField({
+            getValue: () => this.state.location,
+            parse: (value) => this.parseLatLng(value),
+            setValue: (value) => {
+                this.state.location = value;
+                this.placeMarker(value);
+            },
+        });
+
+        onWillStart(() =>
+            Promise.all([
+                loadJS("/web_leaflet_lib/static/lib/leaflet/leaflet.js"),
+                loadCSS("/web_leaflet_lib/static/lib/leaflet/leaflet.css"),
+            ])
+        );
+
+        onMounted(() => {
+            this.initializeMap();
+        });
+
+        onWillUpdateProps(() => {
+            this.placeMarker(this.parseLatLng(this.props.record.data[this.props.name]));
+            this.state.location = this.parseLatLng(
+                this.props.record.data[this.props.name]
+            );
+        });
+
+        onPatched(() => {
+            this.placeMarker(this.parseLatLng(this.props.record.data[this.props.name]));
+            this.resetZoom(this.parseLatLng(this.props.record.data[this.props.name]));
+        });
+
+        onWillPatch(() => {
+            this.placeMarker(this.parseLatLng(this.props.record.data[this.props.name]));
+            this.resetZoom(this.parseLatLng(this.props.record.data[this.props.name]));
+        });
+    }
+
+    initializeMap() {
+        const mapOptions = {
+            center: {lat: this.state.location.lat, lng: this.state.location.lng},
+            zoom: 13,
+        };
+        // eslint-disable-next-line no-undef
+        this.map = L.tileLayer("https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png", {
```

Comment:
Hi could you reuse settings introduced by web_leaflet_lib module ? 

See : 

- https://github.com/OCA/geospatial/blob/16.0/web_view_leaflet_map/static/src/js/view/map/map_renderer.js#L15-L16

- https://github.com/OCA/geospatial/blob/16.0/web_view_leaflet_map/static/src/js/view/map/map_renderer.js#L169-L172


values are set here : https://github.com/OCA/geospatial/pull/383/files#diff-5be4d3d480aee272fac8a7719d7470286c43ba57485ca6fc98da0cc374ee8910R11

Thanks !